core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java

```diff
@@ -212,17 +247,22 @@ public void testCrossJoinBetweenDF()
     }
 
     @Test
-    public void testCrossJoinInequalityNoDFWithCast()
+    public void testCrossJoinInequalityDFWithCastOnTheLeft()
     {
         assertPlan("SELECT o.comment, l.comment FROM lineitem l, orders o WHERE o.comment < l.comment",
                 anyTree(filter("CAST(L_COMMENT AS varchar(79)) > O_COMMENT",
                         join(
                                 INNER,
                                 ImmutableList.of(),
-                                tableScan("lineitem", ImmutableMap.of("L_COMMENT", "comment")),
+                                filter(TRUE_LITERAL,
```

Comment:
assert that dynamic filter is now present (that is the other `join` override)

Reply:
done.
I had to manually create the case expression because SqlParser produces a different expression than what is used in a plan (typeOnly flag differs).